src/js/components/AppListFilterComponent.jsx

```diff
@@ -81,7 +62,8 @@ var AppListFilterComponent = React.createClass({
   },
 
   handleClearFilterText: function () {
-    this.setQueryParam("");
+    this.setQueryParam("filterText", "");
```

Comment:
After merging the other PR into this, it will magically be like you requested. :smile: